subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -2545,6 +2604,151 @@ void ull_conn_default_tx_time_set(uint16_t tx_time)
 }
 #endif /* CONFIG_BT_CTLR_DATA_LENGTH */
 
+#if defined(CONFIG_BT_CTLR_SYNC_TRANSFER_SENDER)
+static bool ticker_op_id_match_func(uint8_t ticker_id, uint32_t ticks_slot,
+				    uint32_t ticks_to_expire, void *op_context)
+{
+	ARG_UNUSED(ticks_slot);
+	ARG_UNUSED(ticks_to_expire);
+
+	uint8_t match_id = *(uint8_t *)op_context;
+
+	return ticker_id == match_id;
+}
+
+static void ticker_get_offset_op_cb(uint32_t status, void *param)
+{
+	*((uint32_t volatile *)param) = status;
+}
+
+static uint32_t get_ticker_offset(uint8_t ticker_id, uint16_t *lazy)
+{
+	uint32_t volatile ret_cb;
+	uint32_t ticks_to_expire;
+	uint32_t ticks_current;
+	uint32_t remainder_us;
+	uint32_t remainder;
+	uint32_t start_us;
+	uint32_t ret;
+	uint8_t id;
+
+	id = TICKER_NULL;
+	ticks_to_expire = 0U;
+	ticks_current = 0U;
+
+	ret_cb = TICKER_STATUS_BUSY;
+
+	ret = ticker_next_slot_get_ext(TICKER_INSTANCE_ID_CTLR, TICKER_USER_ID_ULL_LOW,
+				       &id, &ticks_current, &ticks_to_expire, &remainder,
+				       lazy, ticker_op_id_match_func, &ticker_id,
+				       ticker_get_offset_op_cb, (void *)&ret_cb);
+
+	if (ret == TICKER_STATUS_BUSY) {
+		while (ret_cb == TICKER_STATUS_BUSY) {
+			ticker_job_sched(TICKER_INSTANCE_ID_CTLR, TICKER_USER_ID_ULL_LOW);
+		}
```

Comment:
This approach is done several places in the controller and to get the most accurate and correct answer @cvinayak is probably the right one to answer this. :)

Reply:
This is a busy wait implementation for an interrupt (software interrupt) being triggered but to be enabled by another interrupt (radio interrupt). This is in case of nrf52 never entering the if-clause as the software interrupt is never disabled. This could only be a forever loop for nrf51, but that the case where a radio interrupt never arrives in which case its a h/w issue anyway.